src/Mocking.jl

```diff
@@ -1,39 +1,36 @@
 module Mocking
 
-export @patch, @mock, Patch, apply,
-    DISABLE_COMPILED_MODULES_STR, DISABLE_COMPILED_MODULES_CMD
+export @patch, @mock, Patch, apply
 
 include("expr.jl")
 include("dispatch.jl")
-include("options.jl")
 include("patch.jl")
 include("mock.jl")
 include("deprecated.jl")
 
-# When ENABLED is false the @mock macro is a noop.
-global ENABLED = false
-global PATCH_ENV = nothing
-
-function enable(; force::Bool=false)
-    ENABLED::Bool && return  # Abend early if enabled has already been set
-    global ENABLED = true
-    global PATCH_ENV = PatchEnv()
-
-    if compiled_modules_enabled()
-        if force
-            # Disable using compiled modules when Mocking is enabled
-            set_compiled_modules(false)
-        else
-            @warn(
-                "Mocking.jl will probably not work when $COMPILED_MODULES_FLAG is ",
-                "enabled. Please start `julia` with `$DISABLE_COMPILED_MODULES_STR` ",
-                "or alternatively call `Mocking.enable(force=true).`",
-            )
-        end
-    end
+# Create the initial definition of `activated` which defaults Mocking to be disabled
+activated() = false
+
+"""
+    Mocking.activate()
+
+Enable `@mock` call sites to allow for calling patches instead of the original function.
+"""
+function activate()
+    # Avoid redefining `activated` when it's already set appropriately
+    !activated() && Core.eval(@__MODULE__, :(activated() = true))
```

Comment:
Isn't this the same as:

```suggestion
    !activated() && @eval(activated() = true))
```


(It would be different if `activate()` was a macro, as then it would define the name in the module it was called from, but it isn't)